velox/core/SimpleFunctionMetadata.h

```diff
@@ -186,8 +186,14 @@ struct TypeAnalysis {
         CppToType<T>::isPrimitiveType ||
         CppToType<T>::typeKind == TypeKind::OPAQUE);
     results.stats.concreteCount++;
-    results.out << boost::algorithm::to_lower_copy(
-        std::string(CppToType<T>::name));
+    if (isDecimalKind(CppToType<T>::typeKind)) {
+      results.out << boost::algorithm::to_lower_copy(
+                         std::string(CppToType<T>::name))
```

Comment:
Simply use `DECIMAL(a_precision, a_scale)`. This will avoid supporting `std::string(CppToType<T>::name)`

Reply:
At the time of binding VELOX must exactly match argument types. Setting signature to DECIMAL(a, b) would map LONG_DECIMAL argument type to SHORT_DECIMAL.

Reply:
This problem can happen with Vector functions too then?